xrmocap/core/estimation/mview_mperson_topdown_estimator.py

```diff
@@ -195,66 +216,130 @@ def run(
-                    mframe_keypoints2d = keypoints2d_list[view_idx]
-                    keypoints2d = Keypoints(
-                        kps=mframe_keypoints2d.get_keypoints()[
-                            sub_fidx:sub_fidx + 1, ...],
-                        mask=mframe_keypoints2d.get_mask()[sub_fidx:sub_fidx +
-                                                           1, ...],
-                        convention=mframe_keypoints2d.get_convention(),
-                        logger=self.logger)
-                    n_kps = keypoints2d.get_keypoints_number()
-                    sframe_bbox_list.append(bbox_list[view_idx][sub_fidx])
-                    sframe_keypoints2d_list.append(keypoints2d)
-                association_results, predict_keypoints3d, identities = \
+                    sview_kps2d_idx = []
+                    for idx, bbox2d in enumerate(
+                            bbox2d_list[view_idx][sub_fidx]):
+                        if bbox2d[-1] > self.bbox_thr:
+                            sview_kps2d_idx.append(idx)
+                    sview_kps2d_idx = np.array(sview_kps2d_idx)
+                    if len(sview_kps2d_idx) > 0:
+                        sframe_bbox2d_list.append(
+                            torch.tensor(bbox2d_list[view_idx][sub_fidx])
+                            [sview_kps2d_idx])
+                        mframe_keypoints2d = keypoints2d_list[view_idx]
+                        keypoints2d = Keypoints(
+                            kps=mframe_keypoints2d.get_keypoints()[
+                                sub_fidx:sub_fidx + 1, sview_kps2d_idx],
+                            mask=mframe_keypoints2d.get_mask()[
+                                sub_fidx:sub_fidx + 1, sview_kps2d_idx],
+                            convention=mframe_keypoints2d.get_convention(),
+                            logger=self.logger)
+                        if keypoints2d.get_convention(
+                        ) != self.pred_kps3d_convention:
+                            keypoints2d = convert_keypoints(
+                                keypoints=keypoints2d,
+                                dst=self.pred_kps3d_convention,
+                                approximate=True)
+                        sframe_keypoints2d_list.append(keypoints2d)
+                    else:
+                        sframe_bbox2d_list.append(torch.tensor([]))
+                        sframe_keypoints2d_list.append(
+                            Keypoints(convention=self.pred_kps3d_convention))
+                # Establish cross-frame and cross-person associations
+                sframe_association_results, predict_keypoints3d, identities = \
                     self.associator.associate_frame(
-                        mview_img_arr=mview_batch_arr[:, sub_fidx, ...],
-                        mview_bbox2d=sframe_bbox_list,
+                        mview_img_arr=mview_batch_arr[:, sub_fidx].transpose(
```

Comment:
We could add a commet here: Dimension definition varies between cv2 images and tensor images.